### ISSUE TITLE
Surface error code meaning in documentation & exception messages

### ISSUE THREAD

**Issue by softwareengineerprogrammer** [NREL/python-geophires-x#15]
Currently a number of error codes exist but I am unable to find any 'official' documentation of them other than comments in the code, such as here for error code 5500: https://github.com/NREL/python-geophires-x/blob/b31f3b3906a49cd9d98154e29ecb5ab632815682/src/geophires_x/AGSSurfacePlant.py#L599-L601

Resolution:
1. Identify whether there is existing documentation of error codes -> error causes beyond comments in the existing codebase - @malcolm-dsider or @kfbeckers. (My assumption is that there is not but I would like to confirm this to avoid accidental duplication of effort/canonical source)
1. Surface human-readable error messages in addition to codes where applicable (unit test in https://github.com/NREL/python-geophires-x/blame/b31f3b3906a49cd9d98154e29ecb5ab632815682/tests/test_geophires_x.py#L276 is relevant)

**Comment by softwareengineerprogrammer** [comment]
https://github.com/NREL/GEOPHIRES-X/blob/10694bf6b1000a730ff9d378301aab60a2bf8a3e/src/geophires_x/AGSEconomics.py#L240-L251

**Comment by malcolm-dsider** [comment]
I don’t know what this means. It seems to be a highlighted block of code, but I don’t know why it is highlighted.

**Comment by softwareengineerprogrammer** [comment]
@malcolm-dsider that particular block of code is an example of an undocumented error code (6000) being thrown - on line 242: https://github.com/NREL/GEOPHIRES-X/blob/10694bf6b1000a730ff9d378301aab60a2bf8a3e/src/geophires_x/AGSEconomics.py#L242

**Comment by malcolm-dsider** [comment]
Yep, OK, I can see that. Throwing error code = 6000…. This means the target generation option is electricity, but the system is producing no electricity.

 

It looks like 5000 is undocumented also… that is the case where the generation option is heat, but no heat is produced.

 

How do I document the error codes?

 

I note that there is a code section labeled “handle errors” immediately following these two errors. I note further that the error handler simply writes a generic message and then calls sys.exit(), which is no longer a favored strategy for handling this situation. Should I go looking for a place where you replaced a sys,exit() and use that as a go-by to fix this code?

 

And should I do this before we fix my git rebasing issue? Or wait?

 

-Malcolm

**Comment by softwareengineerprogrammer** [comment]
This can wait - it's low priority and doesn't have a clearly defined
solution. Don't worry about it for now
